vmdb/app/models/tag.rb

```diff
@@ -104,4 +107,31 @@ def self.filter_ns(tags, ns)
   def ==(comparison_object)
     super || name.downcase == comparison_object.to_s.downcase
   end
+
+  def category
+    @category ||= Classification.find_by_name(name_path.split('/').first, nil)
+  end
+
+  def show
+    category.try(:show)
+  end
+
+  def categorization
+    @categorization ||= if !show
+                          {}
+                        else
+                          {
+                            "name"         => classification.name,
+                            "description"  => classification.description,
+                            "category"     => {"name" => category.name, "description" => category.description},
+                            "display_name" => "#{category.description}: #{classification.description}"
+                          }
+                        end
```

Comment:
I personally don't like the alignment way to the right like that...if you move the if to the next line, then you can just two-space indent.